lib/camp.js

```diff
@@ -773,6 +774,38 @@ function pathLikeUnit(serverMethod) {
 var pathUnit = pathLikeUnit('path');
 var notFoundUnit = pathLikeUnit('notFound');
 
+// HTTP method unit (express-like, e.g. `camp.post('/', fn)`).
+function methodUnit (server) {
+  var paths = {};
+
+  function addMethod (method, path, handler) {
+    if (path[0] === '/') {
+      path = path.slice(1);
+    }
+    if (!(path in paths)) {
+      paths[path] = {};
+      server._method(path, function (req, res, next) {
+        var cb = paths[path][req.method.toLowerCase()];
```

Comment:
Indeed, `server._method` is implemented as a `pathLikeUnit`, so parametrized path matching is handled one layer above this. This `cb` holder is just there to match methods.